op-node/node/node.go

```diff
@@ -448,6 +451,7 @@ func (n *OpNode) initHeartbeat(cfg *Config) {
 	if cfg.P2P.Disabled() {
 		peerID = "disabled"
 	} else {
+		// Is there a check for p2p enabled missing here? Is it implied that p2p is enabled if there's a heartbeat?
 		peerID = n.P2P().Host().ID().String()
 	}
```

Comment:
To answer your question, yes I think it is guaranteed because `initHeartbeat` is called after `initP2P`, which sets up  p2p if it's enabled.

I think this check here is actually missing a `nil` check for the `cfg.P2P` field. If you retain the config in the `OpNode` struct and add ` && !n.cfg.P2P.Disabled()` to `OpNode.p2pEnabled()`, then you could use `p2pEnabled()` instead. Or use `cfg.P2PEnabled()` like I suggest in another comment.

Reply:
Thanks, I looked into it a bit further and tightened it up a bit. Looks like the p2p nil check was missed.